spring-kafka/src/main/java/org/springframework/kafka/listener/KafkaMessageListenerContainer.java

```diff
@@ -470,6 +470,7 @@ private void stopInvokerAndCommitManualAcks() {
 			}
 			// handle the last manual acks, after the listeners have closed
 			handleManualAcks();
+			processCommits();
```

Comment:
OK. Never mind. With the `AckMode.RECORD` `processCommits()` doesn't perform `commitIfNecessary()`.
Merging...